File: eMaestroD.NgUI/src/app/Invoices/Models/gl.ts
export interface Gl {

    // GLID?: any;
    // txID ?: 0;
    // relTxID ?: 0;
    // relacctName ?: "";
    // acctName ?: "";
    // relCOAID ?: 0;
    // postTypeID ?: 0;
    // journalID ?: 0;
    // txTypeID ?: any;
    // invoiceID ?: 0;
    // depositID ?: 0;
    // salesManID ?: 0;
    // bookerID ?: 0;
    // locID ?: 0;
    // locName ?: "";
    // empID ?: 0;
    // empFName ?: "";
    // empLName ?: "";
    // cstID ?: 0;
    // cstName ?: "";
    // vendID ?: 0;
    // vendName ?: "";
    // prodID ?: 0;
    // prodName ?: "";
    // prodUnit ?: "";
    // prodCode ?: "";
    // sellRate ?: 0;
    // purchRate ?: 0;
    // avgTp ?: 0;
    // comPrice ?: 0;
    // comID ?: 0;
    // CompanyName ?: "";
    // prodGrpID ?: 0;
    // ProdCategorey ?: "";
    // qty ?: 0;
    // qtyBal ?: 0;
    // unitPrice ?: 0;
    // bonusQty ?: 0;
    // bonusSum ?: 0;
    // instituteOffer ?: 0;
    // creditOffer ?: 0;
    // tradeOffer ?: 0;
    // acctBal ?: 0;
    // balSum ?: 0;
    // creditSum ?: 0;
    // debitSum ?: 0;
    // discountSum ?: 0;
    // paidSum ?: 0;
    // taxSum ?: 0;
    // unusedSum ?: 0;
    // voidedSum ?: 0;
    // dtTx ?: Date;
    // dtDue ?: Date;
    // glComments ?: "";
    // isCleared ?: any;
    // isDeposited ?: any;
    // isPaid ?: any;
    // isVoided ?: any;
    // checkName ?: "";
    // batchNo ?: any;
    // expiry ?: Date;
    // claim ?: 0;
    // checkAdd ?: "";
    // checkNo ?: "";
    // voucherID ?: "";
    // voucherNo ?: any;
    // active ?: true;
    // crtBy ?: "";
    // crtDate ?: Date;
    // modBy ?: "";
    // modDate ?: Date;
    // dtStart ?: Date;
    // dtEnd ?: Date;
    GLID?: any;
    txID?: any;
    txTypeID ?: any,
    prodName ?: any,
    cstName ?: any,
    locID?:any,
    expiry?:any,
    dtDue?:any,
    cstID ?: any,
    isDeposited ?: any,
    isVoided ?: any,
    isCleared ?: any,
    isPaid ?: any,
    voucherNo ?: any,
    creditSum ?: any,
    discountSum ?: any,
    taxSum ?: any,
    paidSum ?: any,
    dtTx ?: any,
    empID ?: any,
    glComments ?: any,
    crtBy ?: any,
    prodID ?: any,
    batchNo ?: any,
    qty?:any,
    bonusQty?:any,
    unitPrice?:any,
    depositID?:any,
    modBy?:any,
    checkName?:any,
    active?:any,
    COAID ?: any;
    coaid ?: any;
    type?:any;
    purchRate?:any;
    crtDate ?: any;
    relCOAID?: any;
    vendID ?: any;
    prodCode?: any;
    debitSum?: any;
    taxName?:any;
    comID?:any;
    qtyBal?: any;
    balSum?:any;
    checkNo?:any;
    checkAdd?:any;
    acctNo?:any;
}

export interface type{
    name?:any;
}

File: Models/Models/GL.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eMaestroD.Models.Models
{
    public class GL
    {
        public int GLID { get; set; }

        public int txID { get; set; }
        public int? comID { get; set; }

        public int relTxID { get; set; }

        public int COAID { get; set; }


        public int relCOAID { get; set; }


        public int postTypeID { get; set; }

        public int txTypeID { get; set; }

        public int invoiceID { get; set; }

        public int depositID { get; set; }

        public int salesManID { get; set; }

        public int bookerID { get; set; }

        public int locID { get; set; }


        public int empID { get; set; }


        public int cstID { get; set; }


        public int vendID { get; set; }


        public int prodID { get; set; }
        public int journalID { get; set; }

        public decimal qty { get; set; }

        public decimal qtyBal { get; set; }

        public decimal unitPrice { get; set; }

        public decimal bonusQty { get; set; }

        public decimal bonusSum { get; set; }

        public decimal instituteOffer { get; set; }

        public decimal creditOffer { get; set; }

        public decimal tradeOffer { get; set; }

        public decimal acctBal { get; set; }

        public decimal balSum { get; set; }

        public decimal creditSum { get; set; }

        public decimal debitSum { get; set; }

        public decimal discountSum { get; set; }
        public decimal extraDiscountSum { get; set; }
        public decimal paidSum { get; set; }
        public decimal taxSum { get; set; }
        public decimal unusedSum { get; set; }
        public decimal voidedSum { get; set; }
        public DateTime? dtTx { get; set; }
        public DateTime? dtDue { get; set; }
        public string? glComments { get; set; }

        [NotMapped]
        public string? prodName { get; set; }
        [NotMapped]
        public string? cstName { get; set; }

        [NotMapped]
        public string? vendName { get; set; }
        [NotMapped]
        public string? taxName { get; set; }

        [NotMapped]
        public string? prodCode { get; set; }
        public bool? isCleared { get; set; }

        public bool? isDeposited { get; set; }

        public bool? isPaid { get; set; }

        public bool? isVoided { get; set; }

        public string? checkName { get; set; }

        public string? batchNo { get; set; }

        public DateTime? expiry { get; set; }

        public decimal claim { get; set; }

        public string? checkAdd { get; set; }

        public string? checkNo { get; set; }

        public string? voucherID { get; set; }

        public string? voucherNo { get; set; }

        public bool? active { get; set; }

        public string? crtBy { get; set; }

        public DateTime? crtDate { get; set; }

        public string? modBy { get; set; }
        public DateTime? modDate { get; set; }
        public string? acctNo { get; set; }
        public string? relAcctNo { get; set; }


        [NotMapped]
        public string? type { get; set; }

        [NotMapped]
        public decimal purchRate { get; set; }
        public object Clone(GL gl)
        {
            return this;
        }

        public int prodBCID { get; set; }
        public decimal rebateSum { get; set; }
        public bool? isConverted { get; set; }

        public List<GLDetail>? gLDetails { get; set; } = new List<GLDetail>();
    }
}
